### ISSUE TITLE
BERT Pretraining Learning Rate Schedule

### ISSUE THREAD

**Issue by brandenchan** [google-research/bert#586]
BERT should be pretrained in 2 phases - 90% of training is done with sequence length 128 and 10% is done with sequence length 512 (c.f. "Pre-training tips and caveats" in README.md).

The learning rate is scheduled to linearly ramp up at the beginning (num_warmup_steps) and then decay over the course of pretraining. (https://github.com/google-research/bert/blob/d66a146741588fb208450bde15aa7db143baaa69/optimization.py#L31-L38)

Does anyone have any suggestions or experience with how the learning rate should be set in the second phase of training when max_seq_len=512? How many warm up steps did you use? Did you set the learning rate to the same value as in the first phase of training?

**Comment by LiweiPeng** [comment]
Yang You etc's paper (Large Batch Optimization for Deep Learning: Training BERT in 76 minutes, https://arxiv.org/abs/1904.00962) has a good discussion on this topic.
